src/main/java/uk/nhs/prm/e2etests/service/SqsService.java

```diff
@@ -1,30 +1,131 @@
+import software.amazon.awssdk.services.sqs.model.DeleteMessageBatchRequest;
+import software.amazon.awssdk.services.sqs.model.DeleteMessageRequest;
+import software.amazon.awssdk.services.sqs.model.Message;
+import software.amazon.awssdk.services.sqs.model.MessageAttributeValue;
+import software.amazon.awssdk.services.sqs.model.PurgeQueueRequest;
+import software.amazon.awssdk.services.sqs.model.QueueAttributeName;
+import software.amazon.awssdk.services.sqs.model.ReceiveMessageRequest;
+import uk.nhs.prm.e2etests.exception.MaxQueueEmptyResponseException;
+import software.amazon.awssdk.services.sqs.model.SendMessageRequest;
+import software.amazon.awssdk.services.sqs.model.SqsException;
+import uk.nhs.prm.e2etests.enumeration.MessageType;
 import uk.nhs.prm.e2etests.exception.ServiceException;
 import uk.nhs.prm.e2etests.model.SqsMessage;
 
+import java.util.stream.Collectors;
+import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
-import java.util.stream.Collectors;
 
+import static uk.nhs.prm.e2etests.utility.ThreadUtility.sleepFor;
+
+@Log4j2
 @Service
 public class SqsService {
+    // Constants
+    private static final int SQS_EMPTY_RESPONSE_LIMIT = 10;
     private static final int MAX_VISIBILITY_TIMEOUT = 43200;
+    private static final int INITIAL_DELAY_MILLISECONDS = 30000;
 
+    // Beans
     private final SqsClient sqsClient;
 
     @Autowired
     public SqsService(AwsCredentialsProvider awsCredentialsProvider) {
         this.sqsClient = SqsClient.builder()
                 .region(Region.EU_WEST_2)
-                .credentialsProvider(awsCredentialsProvider)
-                .build();
+                .credentialsProvider(awsCredentialsProvider).build();
+    }
+
+    /**
+     * This method has a complexity of O(N), where N is
+     * expectedNumberOfEhrCores + expectedNumberOfFragments,
+     * the total number of messages to be received and filtered.
+     * @param queueUri The URI of the Amazon SQS Queue.
+     * @param expectedNumberOfEhrCores The number of cores we are expecting.
+     * @param expectedNumberOfEhrFragments The number of fragments we are expecting.
+     * @param outboundConversationIds The outboundConversationIds.
+     * @return If all the messages were found successfully.
+     */
+    public boolean getAllMessagesFromQueue(int expectedNumberOfEhrCores,
+                                           int expectedNumberOfEhrFragments,
+                                           List<String> outboundConversationIds,
+                                           String queueUri) {
+        final CharSequence[] filterCriteria = outboundConversationIds.toArray(new CharSequence[0]);
+        int totalNumberOfMessages = expectedNumberOfEhrCores + expectedNumberOfEhrFragments;
+        final List<Message> allMessages = new ArrayList<>();
+        boolean allMessagesFound = false;
+
+        int emptyResponseCount = 0;
+        int ehrCoreCount = 0;
+        int ehrFragmentCount = 0;
```

Comment:
do you reckon we rename these to ehrCoreFound and ehrFragmentFound  to be inline with `allMessagesFound` and 'messagesFound' ?